File: 1190. Reverse Substrings Between Each Pair of Parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseParentheses = function(s) {
    let stk = [];

    for(c of s){
        if(c!=')'){
            stk.push(c);
        }else{
            let store = [];

            while(stk[stk.length-1]!='('){
                store.push(stk.pop());
            }
            stk.pop();

            for(let i=0;i<store.length;i++){
                stk.splice(stk.length, 0, store[i]);
            }
        }
    }
    return stk.join('');
};

File: 1190. Reverse Substrings Between Each Pair of Parentheses.py
class Solution:
    def reverseParentheses(self, s: str) -> str:
        stk = []
        
        for c in s:
            if c!=')':
                stk.append(c)
            
            else:
                store = []
                
                while stk[-1]!='(':
                    store.append(stk.pop())
                
                stk.pop();stk.extend(store)
        
        return ''.join(stk)